File: case_studies/m/boost_converter.m
function xdot = dynamics(t, x, u)
    L = 0.001; C = 0.0001; R = 10; Vin = 12;
    iL = x(1); vC = x(2);
    D = max(0, min(1, u));  % duty cycle
    xdot = [(Vin - D*vC) / L;
            (D*iL - vC/R) / C];
end

File: case_studies/py/boost_converter.py
import numpy as np

def dynamics(t, x, u):
    L, C, R, Vin = 0.001, 0.0001, 10, 12
    iL, vC = x[0], x[1]
    D = np.clip(u, 0, 1)
    return np.array([(Vin - D*vC) / L, (D*iL - vC/R) / C])